cishouseholds/validate.py

```diff
@@ -148,3 +150,44 @@ def check_singular_match(
             failure_column_name, F.when(F.col(existing_failure_column) == 1, 1).otherwise(F.col(failure_column_name))
         )
     return df.drop("TOTAL")
+
+
+class ConfigError(Exception):
+    pass
+
+
+def upfront_key_value_parameters_validation(all_function_dict: Dict, config_file_arguments_list: List):
+    """
+    Checks that there's a valid input in the pipeline_config.yaml for every stage
+    input argument.
+
+    Parameters
+    ----------
+    all_function_dict: dictionary of all functions name and function object in pipeline_stages.py
+    pipeline_stage_list: from the config file all the functions that have been set up to run.
+    """
+    error_msg = ""
+    for stage_dict in config_file_arguments_list:
+        if type(stage_dict["run"]) != bool:
+            error_msg += f"Run parameter in {stage_dict['function']} has to be boolean type."
+    if error_msg:
+        raise ConfigError(error_msg)
+
+    config_file_arguments_list_true = [true_stage for true_stage in config_file_arguments_list if true_stage["run"]]
+    for function_run_dict in config_file_arguments_list_true:
+        function_run_list = [x for x in function_run_dict.keys() if (x != "run") and (x != "function")]
```

Comment:
There is an optional "when" parameter that also needs validating - you should check with @slamothe99 what criteria need to be met for that parameter

It will need including in this exclusion set of parameters